src/libraries/Common/src/Interop/Windows/SspiCli/SSPIAuthType.cs

```diff
@@ -62,16 +62,18 @@ public int InitializeSecurityContext(ref SafeFreeCredentials? credential, ref Sa
 
         public int EncryptMessage(SafeDeleteContext context, ref Interop.SspiCli.SecBufferDesc inputOutput, uint qop)
         {
+            bool mustRelease = false;
             try
             {
-                bool ignore = false;
-
-                context.DangerousAddRef(ref ignore);
+                context.DangerousAddRef(ref mustRelease);
```

Comment:
Would the `DangerousAddRef` throw if it fails? 
https://github.com/dotnet/runtime/blob/6c09018af4f268460451face1d9f001c1f791d6d/src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/SafeHandle.cs#L172

There is internal overload that suggest that. 

https://github.com/dotnet/runtime/blob/6c09018af4f268460451face1d9f001c1f791d6d/src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/SafeHandle.cs#L187-L192

It seems like we can simply call it outside of the `try` ... or eat the ODE and return failure.